libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -261,11 +266,11 @@ void WaveClip::OnProjectTempoChange(
       // read-up or signal analysis) we can use something smarter than that. In
       // the meantime, use the tempo of the project when the clip is created as
       // source tempo.
-      mRawAudioTempo = oldTempo;
+      mRawAudioTempo = oldTempo.value_or(newTempo);
 
    if (oldTempo.has_value())
    {
-      const auto ratioChange = *mProjectTempo / newTempo;
+      const auto ratioChange = *oldTempo / newTempo;
```

Comment:
Some objections I raised in earlier commits are resolved here.  It is easily seen that none of the optionals are dereferenced without checking first.